Traceback (most recent call last):
  File "/Users/alinaxu/opt/anaconda3/lib/python3.9/site-packages/jupyter_cache/executors/utils.py", line 58, in single_nb_execution
    executenb(
  File "/Users/alinaxu/opt/anaconda3/lib/python3.9/site-packages/nbclient/client.py", line 1204, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
  File "/Users/alinaxu/opt/anaconda3/lib/python3.9/site-packages/nbclient/util.py", line 84, in wrapped
    return just_run(coro(*args, **kwargs))
  File "/Users/alinaxu/opt/anaconda3/lib/python3.9/site-packages/nbclient/util.py", line 62, in just_run
    return loop.run_until_complete(coro)
  File "/Users/alinaxu/opt/anaconda3/lib/python3.9/asyncio/base_events.py", line 647, in run_until_complete
    return future.result()
  File "/Users/alinaxu/opt/anaconda3/lib/python3.9/site-packages/nbclient/client.py", line 663, in async_execute
    await self.async_execute_cell(
  File "/Users/alinaxu/opt/anaconda3/lib/python3.9/site-packages/nbclient/client.py", line 965, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "/Users/alinaxu/opt/anaconda3/lib/python3.9/site-packages/nbclient/client.py", line 862, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
problearner_parameters = {"splitter":["best","random"], "max_depth" : range(1,50)},
Direct_est = ME_Single(single_dataset, r_model = 'OLS',
                     problearner_parameters = problearner_parameters,
                     truncate = 50, 
                     target_policy=target_policy, control_policy = control_policy, 
                     dim_state = 2, dim_mediator = 1, 
                     MCMC = 50,
                     nature_decomp = True,
                     seed = 10,
                     method = 'Direct')

Direct_est.estimate_DE_ME()
Direct_est.est_DE, Direct_est.est_ME, Direct_est.est_TE,
------------------

[0;31m---------------------------------------------------------------------------[0m
[0;31mValueError[0m                                Traceback (most recent call last)
[0;32m/var/folders/9j/vb5nb4rd5bx0gr1q5ytx9q600000gn/T/ipykernel_64751/2289951171.py[0m in [0;36m<module>[0;34m[0m
[1;32m      1[0m [0mproblearner_parameters[0m [0;34m=[0m [0;34m{[0m[0;34m"splitter"[0m[0;34m:[0m[0;34m[[0m[0;34m"best"[0m[0;34m,[0m[0;34m"random"[0m[0;34m][0m[0;34m,[0m [0;34m"max_depth"[0m [0;34m:[0m [0mrange[0m[0;34m([0m[0;36m1[0m[0;34m,[0m[0;36m50[0m[0;34m)[0m[0;34m}[0m[0;34m,[0m[0;34m[0m[0;34m[0m[0m
[0;32m----> 2[0;31m Direct_est = ME_Single(single_dataset, r_model = 'OLS',
[0m[1;32m      3[0m                      [0mproblearner_parameters[0m [0;34m=[0m [0mproblearner_parameters[0m[0;34m,[0m[0;34m[0m[0;34m[0m[0m
[1;32m      4[0m                      [0mtruncate[0m [0;34m=[0m [0;36m50[0m[0;34m,[0m[0;34m[0m[0;34m[0m[0m
[1;32m      5[0m                      [0mtarget_policy[0m[0;34m=[0m[0mtarget_policy[0m[0;34m,[0m [0mcontrol_policy[0m [0;34m=[0m [0mcontrol_policy[0m[0;34m,[0m[0;34m[0m[0;34m[0m[0m

[0;32m~/opt/anaconda3/lib/python3.9/site-packages/causaldm/learners/CEL/MA/ME_Single.py[0m in [0;36m__init__[0;34m(self, dataset, r_model, problearner_parameters, truncate, target_policy, control_policy, dim_state, dim_mediator, MCMC, seed, nature_decomp, method)[0m
[1;32m     62[0m         self.palearner = PALearner(dataset, problearner_parameters, seed, dim_state = dim_state,
[1;32m     63[0m                                    dim_mediator = dim_mediator)
[0;32m---> 64[0;31m         [0mself[0m[0;34m.[0m[0mpalearner[0m[0;34m.[0m[0mtrain[0m[0;34m([0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m
[0m[1;32m     65[0m [0;34m[0m[0m
[1;32m     66[0m         [0mself[0m[0;34m.[0m[0mpie_A[0m [0;34m=[0m [0mself[0m[0;34m.[0m[0mtarget_policy[0m[0;34m([0m[0mself[0m[0;34m.[0m[0mstate[0m[0;34m,[0m [0mself[0m[0;34m.[0m[0mdim_state[0m[0;34m,[0m [0maction[0m [0;34m=[0m [0mself[0m[0;34m.[0m[0maction[0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m

[0;32m~/opt/anaconda3/lib/python3.9/site-packages/causaldm/learners/CEL/MA/probLearner.py[0m in [0;36mtrain[0;34m(self)[0m
[1;32m    180[0m [0;34m[0m[0m
[1;32m    181[0m         [0mregressor[0m [0;34m=[0m [0mGridSearchCV[0m[0;34m([0m[0mDecisionTreeClassifier[0m[0;34m([0m[0mrandom_state[0m[0;34m=[0m[0mself[0m[0;34m.[0m[0mseed[0m[0;34m)[0m[0;34m,[0m [0mself[0m[0;34m.[0m[0mparameters[0m[0;34m,[0m [0mn_jobs[0m[0;34m=[0m[0;34m-[0m[0;36m1[0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m
[0;32m--> 182[0;31m         [0mregressor[0m[0;34m.[0m[0mfit[0m[0;34m([0m[0mX[0m[0;34m=[0m[0mX[0m[0;34m,[0m [0my[0m[0;34m=[0m[0my[0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m
[0m[1;32m    183[0m         [0mbest_params[0m [0;34m=[0m [0mregressor[0m[0;34m.[0m[0mbest_params_[0m[0;34m[0m[0;34m[0m[0m
[1;32m    184[0m         [0;31m#print('action', best_params)[0m[0;34m[0m[0;34m[0m[0m

[0;32m~/opt/anaconda3/lib/python3.9/site-packages/sklearn/model_selection/_search.py[0m in [0;36mfit[0;34m(self, X, y, groups, **fit_params)[0m
[1;32m    924[0m             [0mrefit_start_time[0m [0;34m=[0m [0mtime[0m[0;34m.[0m[0mtime[0m[0;34m([0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m
[1;32m    925[0m             [0;32mif[0m [0my[0m [0;32mis[0m [0;32mnot[0m [0;32mNone[0m[0;34m:[0m[0;34m[0m[0;34m[0m[0m
[0;32m--> 926[0;31m                 [0mself[0m[0;34m.[0m[0mbest_estimator_[0m[0;34m.[0m[0mfit[0m[0;34m([0m[0mX[0m[0;34m,[0m [0my[0m[0;34m,[0m [0;34m**[0m[0mfit_params[0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m
[0m[1;32m    927[0m             [0;32melse[0m[0;34m:[0m[0;34m[0m[0;34m[0m[0m
[1;32m    928[0m                 [0mself[0m[0;34m.[0m[0mbest_estimator_[0m[0;34m.[0m[0mfit[0m[0;34m([0m[0mX[0m[0;34m,[0m [0;34m**[0m[0mfit_params[0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m

[0;32m~/opt/anaconda3/lib/python3.9/site-packages/sklearn/tree/_classes.py[0m in [0;36mfit[0;34m(self, X, y, sample_weight, check_input, X_idx_sorted)[0m
[1;32m    935[0m         """
[1;32m    936[0m [0;34m[0m[0m
[0;32m--> 937[0;31m         super().fit(
[0m[1;32m    938[0m             [0mX[0m[0;34m,[0m[0;34m[0m[0;34m[0m[0m
[1;32m    939[0m             [0my[0m[0;34m,[0m[0;34m[0m[0;34m[0m[0m

[0;32m~/opt/anaconda3/lib/python3.9/site-packages/sklearn/tree/_classes.py[0m in [0;36mfit[0;34m(self, X, y, sample_weight, check_input, X_idx_sorted)[0m
[1;32m    201[0m [0;34m[0m[0m
[1;32m    202[0m         [0;32mif[0m [0mis_classification[0m[0;34m:[0m[0;34m[0m[0;34m[0m[0m
[0;32m--> 203[0;31m             [0mcheck_classification_targets[0m[0;34m([0m[0my[0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m
[0m[1;32m    204[0m             [0my[0m [0;34m=[0m [0mnp[0m[0;34m.[0m[0mcopy[0m[0;34m([0m[0my[0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m
[1;32m    205[0m [0;34m[0m[0m

[0;32m~/opt/anaconda3/lib/python3.9/site-packages/sklearn/utils/multiclass.py[0m in [0;36mcheck_classification_targets[0;34m(y)[0m
[1;32m    195[0m         [0;34m"multilabel-sequences"[0m[0;34m,[0m[0;34m[0m[0;34m[0m[0m
[1;32m    196[0m     ]:
[0;32m--> 197[0;31m         [0;32mraise[0m [0mValueError[0m[0;34m([0m[0;34m"Unknown label type: %r"[0m [0;34m%[0m [0my_type[0m[0;34m)[0m[0;34m[0m[0;34m[0m[0m
[0m[1;32m    198[0m [0;34m[0m[0m
[1;32m    199[0m [0;34m[0m[0m

[0;31mValueError[0m: Unknown label type: 'continuous'
ValueError: Unknown label type: 'continuous'

